File: ResCheck/src/main/java/com/harrikirik/rescheck/activity/BaseActivity.java
package com.harrikirik.rescheck.activity;

import android.annotation.SuppressLint;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;

import com.harrikirik.rescheck.R;

/**
 * Created by Harri Kirik (harri35@gmail.com)
 */
@SuppressLint("Registered")
public class BaseActivity extends AppCompatActivity {
    protected Toolbar toolbar;

    public Toolbar getToolbar() {
        if (toolbar == null) {
            toolbar = findViewById(R.id.toolbar);
        }
        return toolbar;
    }

    protected void setupToolbar(final String title) {
        setupToolbar(title, false);
    }

    protected void setupToolbar(final String title, @SuppressWarnings("SameParameterValue") final boolean enableUpNavigation) {
        final Toolbar toolbar = getToolbar();
        if (toolbar == null) {
            return;
        }
        toolbar.setTitle(title);
        setSupportActionBar(toolbar);
        if (enableUpNavigation) {
            toolbar.setNavigationIcon(R.drawable.ic_arrow_back_white_24dp);
        }
    }
}

File: ResCheck/src/main/kotlin/com/harrikirik/rescheck/common/BaseActivity.kt
package com.harrikirik.rescheck.common

import android.annotation.SuppressLint
import androidx.fragment.app.Fragment
import androidx.appcompat.app.AppCompatActivity
import com.harrikirik.rescheck.R

@SuppressLint("Registered")
open class BaseActivity : AppCompatActivity() {

    protected fun addFragment(tag: String, newInstance: androidx.fragment.app.Fragment) {
        val fragment = supportFragmentManager.findFragmentByTag(tag)
        if (fragment != null) {
            return
        }
        supportFragmentManager
                .beginTransaction()
                .add(R.id.layout_fragment_container, newInstance, tag)
                .commitAllowingStateLoss()
    }
}
